# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/gmp/ChromiumCDMParent.cpp
# Commit: 6fa7ede8ee41
# Full Hash: 6fa7ede8ee414dc872d5e5c5e68a5c71a63cc7a6
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-08 04:52:24
# Regressor Bug: 1781122
# File Overlap Count: 1
# Description:
#   Bug 1781122 Part 1: Make ChromiumCDMParent mark its Images as DRM. r=media-playback-reviewers,gfx-reviewers,alwu,sotaro
#   
#   This also modifies Image to track whether or not it is from a DRM source.
#   Later parts of the patch move this characteristic into the textures that
#   are used by compositors.
# ==============================================================================

diff -r a2c7bfa9fa8f -r 6fa7ede8ee41 dom/media/gmp/ChromiumCDMParent.cpp
--- a/dom/media/gmp/ChromiumCDMParent.cpp	Wed Sep 07 18:36:06 2022 +0000
+++ b/dom/media/gmp/ChromiumCDMParent.cpp	Wed Sep 07 18:51:11 2022 +0000
@@ -974,6 +974,9 @@
       media::TimeUnit::FromMicroseconds(aFrame.mDuration()), b, false,
       media::TimeUnit::FromMicroseconds(-1), pictureRegion, mKnowsCompositor);
 
+  // This is a DRM image.
+  v->mImage->SetIsDRM(true);
+
   return v.forget();
 }
 